packages/hooks/src/createAddHook.ts

```diff
@@ -1,29 +1,53 @@
 /**
  * Internal dependencies
  */
-import validateNamespace from './validateNamespace.js';
-import validateHookName from './validateHookName.js';
+import validateNamespace from './validateNamespace';
+import validateHookName from './validateHookName';
+import type { Callback, Hooks, StoreKey } from '.';
 
 /**
- * @callback AddHook
  *
  * Adds the hook to the appropriate hooks container.
- *
- * @param {string}               hookName      Name of hook to add
- * @param {string}               namespace     The unique namespace identifying the callback in the form `vendor/plugin/function`.
- * @param {import('.').Callback} callback      Function to call when the hook is run
- * @param {number}               [priority=10] Priority of this hook
  */
+export type AddHook = (
+	/**
+	 * Name of hook to add
+	 */
+	hookName: string,
+	/**
+	 * The unique namespace identifying the callback in the form.
+	 */
+	namespace: string,
+	/**
+	 * Function to call when the hook is run.
+	 */
+	callback: Callback,
+	/**
+	 * Priority of this hook
+	 */
+	priority?: number
+) => void;
+
+export type Handler = {
+	callback: Callback;
+	priority: number;
+	namespace: string;
+};
+
+export type HookInfo = {
+	name: string;
+	currentIndex: number;
+};
```

Comment:
I think we can move this to `types.ts` and remove/replace the `Current` type there with this one.